photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Driver.scala

```diff
@@ -178,12 +178,12 @@ class Driver(val params: Params, val sparkContext: SparkContext, val logger: Pho
     logger.info(s"Number of scored items: $numScoredItems (s)\n")
     val scoredItemsToBeSaved =
       if (numOutputFilesForScores > 0 && numOutputFilesForScores != scoredItems.partitions.length) {
-        scoredItems.coalesce(numOutputFilesForScores)
+        scoredItems.repartition(numOutputFilesForScores)
```

Comment:
Because ```repartition``` will give us exactly the number of partitions specified, while coalesce may not. At the same time, ```repartition``` is easier to test, but has may have larger overhead (extra shuffling) compared to ```coalesce```. 

Since the size of scores here is much smaller than the size of data that need to be "coalesced" in the training driver, and given the tradeoff above, we may want to have ```repartition``` in the scoring driver but ```coalesce``` in the training driver.